### ISSUE TITLE
building libflint.a fails due to missing 3 test file directories

### ISSUE THREAD

**Issue by DanGrayson** [wbhart/flint2#884]
I got the source code by downloading the tar.gz file from the 2.7.0 "release" on github -- for example, it contains no directory called flint2-2.7.0/fq_zech_mpoly/test.

The result was:

```sh
...
make[3]: Leaving directory '/Users/dan/src/M2/M2.git/M2/BUILD/dan/builds.tmp/gallium-release-1.17/libraries/flint/build/flint2-2.7.0/fq_zech_mpoly_factor'
touch: fmpz_mod_mpoly/test/t-*.c: No such file or directory
touch: fq_zech_mpoly/test/t-*.c: No such file or directory
touch: fq_zech_mpoly_factor/test/t-*.c: No such file or directory
make[2]: *** [Makefile:162: libflint.a] Error 1
make[2]: Leaving directory '/Users/dan/src/M2/M2.git/M2/BUILD/dan/builds.tmp/gallium-release-1.17/libraries/flint/build/flint2-2.7.0'
...
```

Complete log attached.
[LOG-1.txt](https://github.com/wbhart/flint2/files/5805176/LOG-1.txt)

**Comment by wbhart** [comment]
Thanks for the report. Interestingly the repo contains no such directories and the tarball on our website is missing those as well, which raises the obvious question: how is our CI passing.

We'll look into this. Thanks.

**Comment by wbhart** [comment]
For now the obvious workaround would be to create those missing directories locally and put some dummy files, e.g. t-test.c in those directories. But this is obviously a pretty severe problem which needs to be fixed, so it may also be better to just wait for 2.7.1 which shouldn't be very far away. There's just this and another issue that we are currently working on and that will happen.

**Comment by DanGrayson** [comment]
> For now the obvious workaround would be to create those missing directories locally and put some dummy files, e.g. t-test.c in those directories. But this is obviously a pretty severe problem which needs to be fixed, so it may also be better to just wait for 2.7.1 which shouldn't be very far away. There's just this and another issue that we are currently working on and that will happen.

Indeed, that's the work-around I've started implementing.

**Comment by wbhart** [comment]
@tthsqe12 Should those dirs be empty? Is it possible some files didn't get committed?

**Comment by tthsqe12** [comment]
Those modules dont have test code becuase they are not complete and are undocumented.

**Comment by wbhart** [comment]
Right, but they require a dummy test file in the directory. Flint won't run its test suite (at least on some machines) without that.

**Comment by tthsqe12** [comment]
Ok. I will add some non trivial test code.

**Comment by wbhart** [comment]
Thanks.

**Comment by DanGrayson** [comment]
It's worth observing that I was making libflint.a, not running tests.

Also, what's the point of touching the test files?  I guess it's to get the tests run again, after having run them previously, but if the dependencies of the tests are correct, then "make" will take care of rerunning the tests when needed.  It might be better to get rid of the "touch" commands.  Or just add a dependency such as "always" so the tests always get run.

**Comment by wbhart** [comment]
@DanGrayson I don't disagree, but this was certainly done for a reason at some point. (And always running the tests is not a viable solution for us.)

The reason I don't want to change this is that every time I make a change which is not absolutely necessary, something else breaks. I prefer to keep the build system as stable as possible unless I can absolutely rule out that the original problem we were working around no longer exists, which unfortunately I can't in this case.

Anyhow, the original issue you reported should be fixed in trunk and the flint-2.7 branch. Needless to say, a 2.7.1 release is not far away. We just have one Windows issue to track down, maybe.

**Comment by wbhart** [comment]
Closing this issue as we believe it to be fixed in 2.7.1 which will be out today.
